src/App.tsx

```diff
@@ -36,22 +36,25 @@ const GlobalStyles = createGlobalStyle`
 
 type AppProps = {
     path: string
+    useBasicGame?: boolean
 }
 
-function App({ path }: AppProps) {
-    const [game, setGame] = useState<GameLogic<WorldState> | null>(null)
+function App({ path, useBasicGame = false }: AppProps) {
+    const [game, setGame] = useState<GameLogic<any> | null>(null)
     useEffect(() => {
         const fetchWorld = async () => {
             const scenarioData = await loadScenario(path)
             if (scenarioData) {
-                const instance = BasicGameScenario.fromData(scenarioData)
+                const instance = useBasicGame
+                    ? BasicGameScenario.fromData(scenarioData)
```

Comment:
Nice to switch implementations with a flag. Don't remember what you said about this though - something we should keep to allow modding? or is this only temporary?

It was related to removing `BasicGameScenario` too right?

Reply:
The intention of having is a flag is to be able to keep both code paths alive and updated until we feel comfortable removing the previous implementation completely.

Reply:
Ah, good idea. It wouldn't hurt to keep the old implementation for some time since we can always use the new one as a default.